### PR TITLE
Adding nightly build

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: cloudbuild/nightly.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+readonly DATE=$(date +%Y%m%d)
+readonly REVISION="0.0.${DATE}"
+readonly MVN="./mvnw -B -e -s /workspace/cloudbuild/gcp-settings.xml -Dmaven.repo.local=/workspace/.repository -Drevision=${REVISION}"
+
+
+cd /workspace
+
+# Build
+$MVN install -DskipTests -Pdsv1_2.12,dsv2
```
Comment (VinayLondhe14) [inline cloudbuild/nightly.sh:33]: Are we not running 2.11 integration tests?

File: cloudbuild/nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# 3. Run unit tests
+  - name: 'gcr.io/$PROJECT_ID/dataproc-spark-bigquery-connector-nightly'
+    id: 'unit-tests'
+    waitFor: ['init']
+    entrypoint: 'bash'
+    args: ['/workspace/cloudbuild/presubmit.sh', 'unittest']
+    env:
+    - 'CODECOV_TOKEN=${_CODECOV_TOKEN}'
+
+  # 4. Run integration tests concurrently with unit tests
```
Comment (VinayLondhe14) [inline cloudbuild/nightly.yaml:25]: Is this comment correct? It says Run integration tests concurrently with unit tests but I see a waitFor block

File: cloudbuild/nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    id: 'integration-tests'
+    waitFor: ['unit-tests']
+    entrypoint: 'bash'
+    args: ['/workspace/cloudbuild/presubmit.sh', 'integrationtest']
+    env:
+      - 'GOOGLE_CLOUD_PROJECT=${_GOOGLE_CLOUD_PROJECT}'
+      - 'TEMPORARY_GCS_BUCKET=${_TEMPORARY_GCS_BUCKET}'
+      - 'CODECOV_TOKEN=${_CODECOV_TOKEN}'
+
+  # 5. Run integration tests concurrently with unit tests
```
Comment (VinayLondhe14) [inline cloudbuild/nightly.yaml:36]: Shouldnt this be acceptance tests?

File: cloudbuild/nightly.sh (hunk 1 above)
Comment (davidrabinowitz) [reply cloudbuild/nightly.sh:33]: Most new usage is done on the 2.12, and we have yet to find an issue that was only on the 2.11

File: cloudbuild/nightly.yaml (hunk 2 above)
Comment (davidrabinowitz) [reply cloudbuild/nightly.yaml]: updated the nightly.yaml

File: cloudbuild/nightly.yaml (hunk 3 above)
Comment (davidrabinowitz) [reply cloudbuild/nightly.yaml]: updated the nightly.yaml

File: cloudbuild/nightly.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+#coverage report
+$MVN test jacoco:report jacoco:report-aggregate -Pcoverage,dsv1_2.12,dsv2
+# Run integration tests
+$MVN failsafe:integration-test failsafe:verify jacoco:report jacoco:report-aggregate -Pcoverage,integration,dsv1_2.12,dsv2_2.4
+# Run integration tests
```
Comment (VinayLondhe14) [inline cloudbuild/nightly.sh:38]: Nit: Shouldn't this be `Run acceptance tests`?

**Review by VinayLondhe14 — APPROVED** [review]
Left a minor comment. Other than that looks good 👍

File: cloudbuild/nightly.sh (hunk 4 above)
Comment (davidrabinowitz) [reply cloudbuild/nightly.sh]: yes

**Pull request merged**